### PR TITLE
fix(route-viewer-overlay): avoid crash on empty but not missing geometry

### PR DESCRIPTION
**Pull request by miles-grant-ibigroup**
This PR resolves crashes found in OTP-RR. We were checking for lack of geometry, but not empty geometry.

### CODE REVIEW

**Review by philip-cline — APPROVED** [review]
Nice fix!

File: packages/route-viewer-overlay/src/index.js [hunk 1]
```diff
@@ -53,19 +53,17 @@ class RouteViewerOverlay extends MapLayer {
   // TODO: determine why the default MapLayer componentWillUnmount() method throws an error
   componentWillUnmount() {}
 
-  componentDidUpdate(prevProps) {
-    // if pattern geometry just finished populating, update the map points
-    if (
-      !isGeomComplete(prevProps.routeData) &&
```
Comment (binh-dam-ibigroup) [inline packages/route-viewer-overlay/src/index.js:59]: Why was this condition removed?

Comment (binh-dam-ibigroup) [reply packages/route-viewer-overlay/src/index.js:59]: Perhaps the condition needs to include checking whether the routeData has changed too.

Comment (miles-grant-ibigroup) [reply packages/route-viewer-overlay/src/index.js:59]: Other than `path` which I don't think we need to expect changing after render all props that can change should trigger a re-center

**Review by binh-dam-ibigroup — APPROVED** [review]

File: packages/route-viewer-overlay/src/index.js [hunk 2]
```diff
@@ -109,6 +107,10 @@ class RouteViewerOverlay extends MapLayer {
 }
 
 RouteViewerOverlay.propTypes = {
+  /**
+   * This boolean value allows disabling of map centering and panning.
+   */
+  allowMapCentering: PropTypes.boolean,
```
[the remark this answers is not in the document]
Comment (miles-grant-ibigroup) [reply packages/route-viewer-overlay/src/index.js]: Nice catch thank you!

**Pull request merged**